### PR TITLE
Added `zindex--navigation` utility classname

### PR DESCRIPTION
**Pull request by tonysight**
## Description
- Fixed typo in some layout classnames in storybook
- Added `zindex--navigation` utility classname. In [ROCKET-726](https://cbinsights.atlassian.net/browse/ROCKET-726?atlOrigin=eyJpIjoiODI3N2VkNGU2ZGI4NDllMDk2NTEyMWRhZjdmM2E1MjAiLCJwIjoiaiJ9), the action bar normally is at `zindex--navigation`, but need to be at the same level of the **Update Table** modal in the case of Stories (`zindex--modal`). So have to toggle between these two zindex values. Also there are a few components using the css variable that can use this new class too.

## Checklist
- [x] Docs have been rebuilt (`yarn build:full`)
- [x] All unit tests pass
- [x] Snapshots have been updated
- [x] No lint errors or warnings have been introduced
- [ ] **[Version bumped](https://github.com/cbinsights/form-design-system#updating-version-number) if appropriate**

### CODE REVIEW

File: src/base-styles/helperClasses/layout/layout.stories.mdx [hunk 1]
```diff
@@ -73,8 +73,8 @@ Sets element width in percentage.
       ['display--block', 'Sets `display` to `block`'],
       ['display--flex', 'Sets `display` to `flex`'],
       ['display--inline', 'Sets `display` to `inline`'],
-      ['display--inline-block', 'Sets `display` to `inline-block`'],
-      ['display--inline-flex', 'Sets `display` to `inline-flex`'],
+      ['display--inlineBlock', 'Sets `display` to `inline-block`'],
```
Comment (rey-wright) [inline src/base-styles/helperClasses/layout/layout.stories.mdx:76]: whoops! Good catch.

**Review by rey-wright — APPROVED** [review]
Thanks kindly for the PR 🙏

**Review by reywright — APPROVED** [review]

**Review by akdetrick — COMMENTED** [review]
Awesome, thanks!

This is a new feature, so I'd recommend [bumping the minor version](https://github.com/cbinsights/form-design-system/#updating-version-number) and entering your change into the `CHAGELOG.md`

**Comment by tonysight** [comment]
@akdetrick just bumped the minor version and added to changelog

**Pull request merged**